### PR TITLE
CID-5635 Migrate repo to GHA for build

### PR DESCRIPTION
**Pull request by cristianmartinello-wf**
CID-5635 Migrate repo to GHA for build

### CODE REVIEW

**Comment by aviary3-wk** [comment]
## Security Insights
No security relevant content was detected by automated scans.

## Action Items
* Review PR for [security impact](https://wiki.atl.workiva.net/display/SECURITY/Development+Security+Review+Guidelines); comment "security review required" if needed or unsure
* Verify [`aviary.yaml`](https://wiki.atl.workiva.net/pages/viewpage.action?pageId=42762752#SecurityInsightFramework&Tooling-ConfiguringAviary) coverage of security relevant code
---
_Questions or Comments? Reach out on Slack: #support-infosec._

**Comment by micahbeeman-wf** [comment]
We will need to bump the version in pubspec.yaml to 0.7.4

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+  pull_request:
+
+permissions:
+  checks: write
+  contents: read
+  id-token: write
+
+jobs:
+  build:
+    runs-on: [self-hosted, small]
```
Comment (micahbeeman-wf) [inline .github/workflows/ci.yml:13]: This needs to run on a github hosted runner

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+
+      - run: |
+          echo "Starting the script sections"
+          dartfmt --set-exit-if-changed -n lib test tool example
+          timeout 5m pub get
+          pub run test
+          tar -czvf unscripted.pub.tgz pubspec.yaml lib README.md LICENSE
+          echo "Script sections completed"
+
+      - uses: Workiva/gha-store-artifacts@refs/tags/v1.0.16
```
Comment (micahbeeman-wf) [inline .github/workflows/ci.yml:27]: we can't use this on a public repo

Comment (cristianmartinello-wf) [reply .github/workflows/ci.yml:27]: This feels like a blocker to the migration of this repo to GHA. Any thoughts @matthewbalvanz-wf ?

**Comment by micahbeeman-wf** [comment]
We have decided to archive this repo instead.

**Pull request closed without merging**